src/thor/bidirectional_astar.cc

```diff
@@ -280,17 +280,12 @@ inline bool BidirectionalAStar::ExpandInner(baldr::GraphReader& graphreader,
                          : costing_->EdgeCost(opp_edge, t2, time_info.second_of_week, flow_sources));
 
   // Separate out transition cost.
-  InternalTurn internal_turn;
-  sif::Cost transition_cost;
-  if (FORWARD) {
-    transition_cost = costing_->TransitionCost(meta.edge, nodeinfo, pred);
-  } else {
-    internal_turn = costing_->TurnType(meta.edge->localedgeidx(), nodeinfo, opp_edge, opp_pred_edge);
-    transition_cost =
-        costing_->TransitionCostReverse(meta.edge->localedgeidx(), nodeinfo, opp_edge, opp_pred_edge,
-                                        static_cast<bool>(flow_sources & kDefaultFlowMask),
-                                        internal_turn);
-  }
+  sif::Cost transition_cost =
+      FORWARD ? costing_->TransitionCost(meta.edge, nodeinfo, pred)
+              : costing_->TransitionCostReverse(meta.edge->localedgeidx(), nodeinfo, opp_edge,
+                                                opp_pred_edge,
+                                                static_cast<bool>(flow_sources & kDefaultFlowMask),
+                                                pred.internal_turn());
```

Comment:
The key is this line https://github.com/valhalla/valhalla/blob/c9ddd4bfa631b7dc44939bc35bde8017159712c2/valhalla/sif/dynamiccost.h#L614
We need to check whether the **previous** maneuver is internal turn and the current one is a turn. And the "previousness" is the same for both forward and reverse searches

Reply:
haha i looked at  this change in the previous pr several times and i wasnt a  fan of the temporary internal_turn variable, im glad to see it go

Reply:
i'm feeling the same 😄